### DISCUSSION TITLE
How to intercept multiple events having same URL

### DISCUSSION THREAD

**Discussion by akhila2589** [cypress-io/cypress#23491] [General]
After a button click multiple xhr requests are happening with same URL. I want to verify whether the payload of one of those requests contains a specific text. How can I do that?

**Comment by jjhelguero** [comment]
If it is a graphQL request, you'll have to use a cy.intercept() with a condition on the operationName.

```js
cy.intercept('POST', '/v1/graphQL', (req) => {
  // check request has property and operation name
  const hasOperationName = req.body.hasOwnProperty('operationName') && body.operationName === 'operation-name'

  if(hasOperationName) {
    // create alias so you can wait on the call
    req.alias = 'operation-name-alias'
  }
})

// button click
cy.wait('operation-name-alias')
```

Docs has [examples](https://docs.cypress.io/guides/end-to-end-testing/working-with-graphql).

**Reply by amyzhao-seismic** [reply]
Hey guys, I met the similar issue, I want to mock API data for different widgets use the same api, the same url, maybe the query id the same but different payload and different response, how can I mock all API data use intercept.
